ShopNow/UserApp/views.py

```diff
@@ -1,134 +1,402 @@
-        else:
-            return Response("Data is not inluded in request.")
-    else:
-        return Response("Invalid method")
-
-
-       
-@csrf_exempt
+                return Response(f"your data is invalid, {serialized.errors}", status = status.HTTP_400_BAD_REQUEST)
+            
 @api_view(['POST'])
-def loginUser(request):
-    if request.method=='POST':
-        if request.data:
-            serialized=loginSerializer(data=request.data)
-            print(request.data)
+def login_user(request):
+            serialized = LoginSerializer(data = request.data)
             if serialized.is_valid():
-                print(serialized.data.get('username'),serialized.validated_data.get('password'))
-                user=authenticate(request,username=serialized.validated_data.get('username'),password=serialized.validated_data.get('password'))
-                print(user)
+                username = serialized.validated_data.get('username')
+                password = serialized.validated_data.get('password')
+                user = authenticate(request, username = username, password = password)
                 if user is not None:
-                    refresh_and_access_token=RefreshToken.for_user(user)
+                    loged_user = Webuser.objects.get(username = username)
+                    logged_user_id = user.id
+                    try:
+                        CartModel.objects.get(user_of_cart = loged_user)
+                    except:
+                        try:
+                            CartModel.objects.create(user_of_cart = loged_user)
+                        except:
+                            return Response("cart is causing some issue")
+                    refresh_and_access_token = RefreshToken.for_user(user)
                     access_token = str(refresh_and_access_token.access_token)
                     refresh_token = str(refresh_and_access_token)
                     response_to_be_send={
-                                'username':serialized.validated_data.get('username'),
-                                'access': access_token,
-                                'refresh':refresh_token,
-                            }
-                    return Response(response_to_be_send)         
+                                        'username':username,
+                                        'access': access_token,
+                                        'refresh':refresh_token,
+                                        'logged_user_id':logged_user_id}
+                    return Response(response_to_be_send, status = status.HTTP_200_OK)
+                else:
+                    return Response("Invalid credentials", status = status.HTTP_404_NOT_FOUND)
             else: 
-                return Response(serialized.errors)      
-             
+                return Response(serialized.errors, status = status.HTTP_400_BAD_REQUEST)      
 
+class RegisteredUsers(APIView):
```

Comment:
I think you should make separate class views and make separate URLs for them, in this way we would not need to check pk condition and code would be more cleaner.

Example:

```
    class RegisteredUsers(APIView):
       authentication_classes = [JWTAuthentication]
       permission_classes = [CustomizeAPIPermissions]

    def get(self, request, pk):
                try:
                    instance = Webuser.objects.get(id = pk)
                    serialized = WebUserSerializer(instance)
                    return Response(serialized.data) 
                except:
                    return Response("User Not exists ", status = status.HTTP_404_NOT_FOUND)
```

```
class RegisteruserListView(APIView):
    def get(self, request):
        users = Webuser.objects.all()
        serialized = WebUserSerializer(users, many=True)
        return Response(serialized.data)

```

`urlpatterns = [
    path('users/', RegisteruserListView.as_view()),        # For listing all users
    path('users/<int:pk>/', RegisteredUsers.as_view()),  # For retrieving a specific user
]`